xarray/backends/api.py

```diff
@@ -414,6 +415,56 @@ def _dataset_from_backend_dataset(
     return ds
 
 
+def _datatree_from_backend_datatree(
+    backend_tree,
+    filename_or_obj,
+    engine,
+    chunks,
+    cache,
+    overwrite_encoded_chunks,
+    inline_array,
+    chunked_array_type,
+    from_array_kwargs,
+    **extra_tokens,
+):
+    if not isinstance(chunks, int | dict) and chunks not in {None, "auto"}:
+        raise ValueError(
+            f"chunks must be an int, dict, 'auto', or None. Instead found {chunks}."
+        )
+
+    # _protect_datatree_variables_inplace(backend_tree, cache)
+    if chunks is None:
+        tree = backend_tree
+    else:
+        tree = DataTree.from_dict(
+            {
+                path: _chunk_ds(
+                    node.dataset,
+                    filename_or_obj,
+                    engine,
+                    chunks,
+                    overwrite_encoded_chunks,
+                    inline_array,
+                    chunked_array_type,
+                    from_array_kwargs,
+                    **extra_tokens,
+                )
+                for path, [node] in group_subtrees(backend_tree)
+            }
+        )
+
+    # ds.set_close(backend_ds._close)
```

Comment:
`backend_tree` should have been created using [`datatree_from_dict_with_io_cleanup`](https://github.com/pydata/xarray/blob/b133fdca71fd614b8056c3e995062b2834320ba5/xarray/backends/common.py#L152), so one way to handle this could be just to copy over the `_close` attribute from every node of `backend_tree`?

Reply:
the question is, do we even need that here? I copied this from `open_dataset` where this is explicitly set, but since `datatree_from_dict_with_io_cleanup` does this already we might be able to just remove it?

The only reason why I kept the commented-out line is to discuss whether the shift in paradigm (have the backend set `_close` vs. do it for all backends the same way) is intentional, and if we should do the same for `open_dataset`.

Reply:
It does not (I think), so I'm explicitly copying it over. So far that doesn't appear to cause anything to break.